R/lint.R

```diff
@@ -182,20 +182,24 @@ lint_dir <- function(path = ".", ...,
     return(lints)
   }
 
-  pb <- if (isTRUE(show_progress)) {
-    txtProgressBar(max = length(files), style = 3L)
+  if (isTRUE(show_progress)) {
+    lints <- lapply(
+      # NB: This cli API is experimental (https://github.com/r-lib/cli/issues/709)
+      cli::cli_progress_along(files, name = "Running linters"),
+      function(idx) {
+        lint(files[idx], ..., parse_settings = FALSE, exclusions = exclusions)
+      }
+    )
+  } else {
+    lints <- lapply(
+      files,
+      function(file) {
```

Comment:
Should I just no lint this?

It's catch-22 situation:

- if I use anonymous function, it triggers `unnecessary_lambda_linter()` lint
- if I don't, it leads to `Error:  The use of linters of class 'function' was deprecated in lintr version 3.0.0` error

Reply:
I decided to nolint it.